source/internal/interactive_batch.cpp

```diff
@@ -0,0 +1,275 @@
+#include "interactive_batch.h"
+#include "interactive_session.h"
+#include "common.h"
+
+#include <functional>
+
+namespace mixer_internal
+{
+
+interactive_batch_entry_internal::interactive_batch_entry_internal()
+	: next(nullptr), value(rapidjson::kObjectType)
+{
+}
+
+interactive_batch_array_internal::interactive_batch_array_internal()
+	: value(rapidjson::kArrayType)
+{
+}
+
+int interactive_batch_begin(interactive_session session, interactive_batch* batchPtr, char *methodName, on_get_batch_params getParams)
```

Comment:
By convention output parameters are always the last parameters so this should be `(interactive_session session, char* methodName, on_get_batch_params getParams, interactive_batch* batchPtr)`